src/CMakeLists.txt

```diff
@@ -129,12 +143,29 @@ else()
     target_sources(umf PRIVATE memory_pool_default.c)
 endif()
 
+# Currently, Level Zero provider is only built for Linux
+if(LINUX AND UMF_BUILD_LEVEL_ZERO_PROVIDER)
+    target_sources(umf PRIVATE
+        provider/provider_level_zero.c)
+
+    # WA for error ze_api.h:14234:20: no newline at end of file [-Werror,-Wnewline-eof]
+    set_source_files_properties(provider/provider_level_zero.c 
```

Comment:
You might want to use: `PROPERTIES APPEND_STRING PROPERTY COMPILE_FLAGS` to append a new flag instead of replacing the existing ones. Ref: https://stackoverflow.com/a/77523503/5935594

Reply:
done